catalog/async/catalog-async-data-api/src/main/java/org/codice/ddf/catalog/async/data/api/internal/ProcessResource.java

```diff
@@ -0,0 +1,88 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.catalog.async.data.api.internal;
+
+import java.io.IOException;
+import java.io.InputStream;
+
+/**
+ * <p>
+ * <b> This code is experimental. While this interface is functional and tested, it may change or be
+ * removed in a future version of the library. </b>
+ * </p>
+ *
+ * <p>
+ * A {@code ProcessResource} represents a local or remote resource that will be used during processing
+ * by the {@link ProcessingFramework}.
+ */
+public interface ProcessResource {
+
+    String DEFAULT_MIME_TYPE = "application/octet-stream";
+
+    String DEFAULT_FILE_NAME = "content_store_file.bin";
+
+    /**
+     * Gets a URI that represents the resource in the catalog.
+     *
+     * @return the URI of the {@link ProcessResource}
+     */
+    String getUri();
+
+    /**
+     * An optional field to provide additional information about a {@link ProcessResource}
+     *
+     * @return the qualifier of the {@link ProcessResource}
+     */
+    String getQualifier();
+
+    /**
+     * The filename of the {@link ProcessResource}. If not set it will default to {@link #DEFAULT_FILE_NAME}
+     *
+     * @return the filename of the {@link ProcessResource}
+     */
+    String getFilename();
```

Comment:
After sleeping on it, I don't think this comment makes a whole lot of sense. However, since the file name is currently only used when creating the `ContentItemImpl` (https://github.com/codice/ddf/pull/1495/files#diff-c5491fd87e13469fa7535a1005fb7c9fR226), and given that `ContentItemImpl` has a constructor that doesn't take a file name, it looks like we could do without this method in the interface.